backend/services/interfaces/ICourseService.ts

```diff
@@ -15,9 +25,59 @@ export interface UpdateCourseRequestDTO {
   description: string;
   image: string;
   previewImage: string;
-  lessons: [string];
   private: boolean;
   published: boolean;
+  modules: { [id: string]: Module };
+}
+
+interface Module {
+  title: string;
+  description: string;
+  image: string;
+  previewImage: string;
+  published: boolean;
+  lessons: string[];
+}
+
+export interface SerializedCreateCourseRequestDTO {
+  title: string;
+  description: string;
+  image: string;
+  previewImage: string;
+  private: boolean;
+  published: boolean;
+  modules: ModuleDTO[];
+}
+
+export interface SerializedUpdateCourseRequestDTO {
+  title: string;
+  description: string;
+  image: string;
+  previewImage: string;
+  private: boolean;
+  published: boolean;
+  modules: ModuleDTO[];
+}
+
+export interface SerializedCourseResponseDTO {
```

Comment:
I wonder if we could extend/partial the regular DTOs, and simply replace the modules attribute with the correct type. Is that something we could do? It would definitely cut down on the duplication of code.

Reply:
We definitely could, I don't really know why I did this tbh